File: ecommerce-webapp/src/app/shared/types/product.ts
import { Author } from './author';
import { Category } from './category';
import { ResponsePageList } from './response';

export type Product = {
  code: string;
  name: string;
  description: string;
  imageUrl: string;
  price: number;
  quantity: number;
  category: Category;
  author: Author;
  tags: string[];
};

export type ProductListResponse = ResponsePageList<Product>;

File: catalog-service/src/main/java/com/qtuan02/catalog/domain/products/Product.java
package com.qtuan02.catalog.domain.products;

import com.qtuan02.catalog.domain.authors.Author;
import com.qtuan02.catalog.domain.categories.Category;
import java.math.BigDecimal;

public record Product(
        String code,
        String name,
        String description,
        String imageUrl,
        BigDecimal price,
        Integer quantity,
        Category category,
        Author author,
        String[] tags) {}
